### DISCUSSION TITLE
Showcase insertion of CMake variables via `configure_file()`

### DISCUSSION THREAD

**Discussion by avitase** [friendlyanon/cmake-init#87] [Ideas]
Occasionally, I have to use CMake variables inside my C/C++ files. A canonical way to get access to these variables is to parse the file via [`configure_file()`](https://cmake.org/cmake/help/latest/command/configure_file.html); however, I find it non-trivial to insert this into the existing template. What about showcasing this by, for example, returning [`CMAKE_PROJECT_NAME`](https://cmake.org/cmake/help/latest/variable/CMAKE_PROJECT_NAME.html) instead of hard coding the name [here](https://github.com/friendlyanon/cmake-init/blob/4dd6e6f81dd332ea3a16971a7974144b05ff5128/cmake-init/templates/c/header/include/__name__/__name__.h#L18). Another common use case is to return the project version as a string.

**Comment by friendlyanon** [comment]
I don't think that's necessary. There are [examples](https://github.com/friendlyanon/cmake-init/wiki/Examples) that use the function, such as [this](https://github.com/friendlyanon/generate-opaque-structs) one.

I would prefer to keep things minimal as a baseline and put interesting things in the wiki as an example.

This might also be better as a discussion rather than an issue I think.

**Comment by avitase** [comment]
Does this example still work for multi-configs scenarios? For example this line looks outdated: https://github.com/friendlyanon/generate-opaque-structs/blob/218bd01cbdaf038d6ddf89bc4f9715ce4de7a782/CMakeLists.txt#L45

**Comment by friendlyanon** [comment]
There is nothing outdated there. I don't understand. `configure_file`'s output happens at configure time, regardless of the config.
